src/OmniSharp.Roslyn.CSharp/Services/BlockStructureWorkspaceOptionsProvider.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Composition;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Options;
+using OmniSharp.Options;
+using OmniSharp.Roslyn.Options;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.CSharp.Services
+{
+    [Export(typeof(IWorkspaceOptionsProvider)), Shared]
+    public class BlockStructureWorkspaceOptionsProvider : IWorkspaceOptionsProvider
+    {
+        private readonly IAssemblyLoader _assemblyLoader;
+        private readonly Lazy<Assembly> _csharpFeatureAssembly;
+        private readonly Lazy<Type> _blockStructureOptions;
+
+        [ImportingConstructor]
+        public BlockStructureWorkspaceOptionsProvider(IAssemblyLoader assemblyLoader)
+        {
+            _assemblyLoader = assemblyLoader;
+            _csharpFeatureAssembly = _assemblyLoader.LazyLoad(Configuration.RoslynFeatures);
+            _blockStructureOptions = _csharpFeatureAssembly.LazyGetType("Microsoft.CodeAnalysis.Structure.BlockStructureOptions");
+        }
+
+        public int Order => 100;
```

Comment:
could you give it a different (unique) order? at the moment we have values 0, 100, 110, 200 already on various providers

the goal was to have them unique so that the applying of options is deterministic

Reply:
Sure thing!